sklearn/impute.py

```diff
@@ -879,7 +907,7 @@ def fit_transform(self, X, y=None):
         self.initial_imputer_ = None
         X, Xt, mask_missing_values = self._initial_imputation(X)
 
-        if self.n_iter == 0:
+        if self.max_iter == 0:
             return Xt
```

Comment:
This line is not covered by a test (as previously). Could you add this case to the unit test?